src/connectors/autocomplete/__tests__/connectAutocomplete-test.js

```diff
@@ -0,0 +1,70 @@
+import jsHelper from 'algoliasearch-helper';
+import connectAutocomplete from '../connectAutocomplete.js';
+
+const fakeClient = { addAlgoliaAgent: () => {} };
+
+describe('connectAutocomplete', () => {
+  it('throws without `renderFn`', () => {
+    expect(() => connectAutocomplete()).toThrow();
+  });
+
+  it('renders during init and render', () => {
+    const renderFn = jest.fn();
+    const makeWidget = connectAutocomplete(renderFn);
+    const widget = makeWidget();
+
+    expect(renderFn).toHaveBeenCalledTimes(0);
+
+    const helper = jsHelper(fakeClient, '', {});
+    helper.search = jest.fn();
+
+    widget.init({
+      helper,
+      instantSearchInstance: {},
+    });
+
+    expect(renderFn).toHaveBeenCalledTimes(1);
+    expect(renderFn.mock.calls[0][1]).toBeTruthy();
+
+    widget.render({
+      widgetParams: {},
+      indices: widget.indices,
+      instantSearchInstance: widget.instantSearchInstance,
+    });
+
+    expect(renderFn).toHaveBeenCalledTimes(2);
+    expect(renderFn.mock.calls[1][1]).toBeFalsy();
+  });
+
+  it('creates derived helper', () => {
+    const renderFn = jest.fn();
+    const makeWidget = connectAutocomplete(renderFn);
+    const widget = makeWidget({ indices: [{ label: 'foo', value: 'foo' }] });
+
+    const helper = jsHelper(fakeClient, '', {});
+    helper.search = jest.fn();
+
+    widget.init({ helper, instantSearchInstance: {} });
+
+    // original helper + derived one
+    expect(widget.indices).toHaveLength(2);
```

Comment:
I'm not sure about relying on a property that is somewhat internal. What about checking that search is called twice if you have two indices?

Reply:
I guess a way to know if the search has been done is by using `widgets.indices[x].helper.search = jest.fn()` but it's still using `widget.indices` do you have any proposal?

Reply:
You can rely on the arguments of the `renderFn`. The `indices` property is part of the public API on this function.

Reply:
Yes if it is something you get from renderFn.